### DISCUSSION TITLE
How to disconnect a socket temporarily

### DISCUSSION THREAD

**Discussion by cachapa** [braverhealth/phoenix-socket-dart#25] [Q&A] [answered]
Is there a way to "pause" a connection, i.e. close the connection temporarily to recover it later?
My intention is to close the connection when my app is no longer in the foreground, and re-open it when the user brings it back.

Calling `close()` disposes the socket which throws an exception on a later `connect()`.
Passing the `reconnect` flag as `true` seems to just cause the socket to reconnect immediately after. I'm not even sure what's the point of it doing that, is it intended behaviour?

**Comment by matehat** [comment, marked as answer]
Currently it's not possible to do that, and I don't think the reference implementation from the core Phoenix team supports that either. A solution would be to close and dispose of the socket when you go in the background, and re-instantiate a new `PhoenixSocket` when your app resumes.

The `reconnect` flag is used internally following a connection close, due to protocol error or transient network failure, for cases where you can get a new connection by re-connecting.

**Reply by cachapa** [reply]
I see, thanks. I was confused by the name of the flag.
I'll go the socket disposal route then, thanks!

**Reply by matehat** [reply]
@cachapa great! don't hesitate if you hit other roadblocks!
